vtex-sites/base.store
# Fix:  slide over scroll behavior

## What's the purpose of this pull request?

This PR aims to fix the scroll behavior while opening some SlideOver related components (Filter, Menu, Minicart) on the mobile version. It also refactors some code so that we can prioritize the use of optional chaining instead of if-clauses.

## How does it work?

1. The first step was to identify what was affecting the scroll behavior: `SlideOver` component while adding the `no-scroll` `class` to the `layout` `div`.
2. Looking into the CSS applied: 
```css
#layout.no-scroll {
  position: fixed;
  top: 0;
  right: 0;
  left: 0;
  overflow: hidden;
}
```
The `position: fixed` makes the `layout div` scroll to the top. So first of all we need to remove this piece of code and keep just the `overflow: hidden;`.

3. But if we remove the `fixed position` on the `layout div`, we are going to have another problem: the scroll from the `body` element while the filter is opened. We need to keep the `body` frozen instead of the `layout div`. So we need to adapt this code to deal with the `body` instead of the `layout div`.

|BEFORE|AFTER|
|-|-|
|![BefaXkYYL8659](https://user-images.githubusercontent.com/11325562/160005538-84e13417-96d1-4b85-9f10-5a120e753e7a.gif)|![AfOuXNS5Q80E](https://user-images.githubusercontent.com/11325562/160005619-a30bd405-28fa-444c-83ae-27099c24bf90.gif)|

## How to test it?

Double-check in the preview link if we have the scroll behavior while opening some `SlideOver` related components (Filter, Menu, Minicart) on the mobile version.

## References

https://vtex-dev.atlassian.net/browse/FSSS-212
https://developer.mozilla.org/pt-BR/docs/Web/JavaScript/Reference/Operators/Optional_chaining
https://medium.com/react-camp/how-to-fight-the-body-scroll-2b00267b37ac
https://futurefoundry.co/blog/disable-body-scroll-ios/

## Checklist

<em>You may erase this after checking them all ;)</em>
